### ISSUE TITLE
choosing-reactor.xhtml references doc links from when doc/examples existed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4384]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#4384|
|Type|defect|
|Created|2010-03-29 00:02:56Z|
|Branch|https://github.com/twisted/twisted/tree/choosing-reactor-links-4384|

<details><summary>Searchable metadata</summary>

```
trac-id__4384 4384
type__defect defect
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__branches_choosing_reactor_links_4384 branches/choosing-reactor-links-4384
branch_author__jesstess jesstess
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1269820976000000 1269820976000000
changetime__1275820341000000 1275820341000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess commented|
|-|-|

(In [28781]) Branching to 'choosing-reactor-links-4384'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_2"></a>|@jesstess commented|
|-|-|

(In [28782]) Fix broken example links and remove mention of the gtkreactor.

refs [#4384](https://github.com/twisted/twisted/issues/4384)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess removed owner|
|-|-|

2833 indicated that it was appropriate to remove mention of gtkreactor. Is there anything else out of date in this file?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @jesstess|
|-|-|

Thanks!

  1. The first hunk in the change is right near a mention of using twistd to run TAPs.  Since it's _so_ near something being changed anyway, maybe you could change this to TACs? :)
  1. Hooray for removing the gtkreactor stuff.
  1. Some other general stuff about this doc:
    1. The overview says that the two general purpose reactor implementations are select and poll.  I think the intent here was to convey the idea that these two reactors work on all platforms.  However, we've since learned that poll doesn't work on OS X.  So really, select is the only general purpose reactor implementation.  It might make sense to clarify the language about "general purpose" by just saying "cross-platform" instead.  And remove poll from the list (now containing one element...).
    1. The "Platform-specific" reactor list could probably be clarified a bit.  KQueue and Epoll look alright (although the KQueue reactor isn't actually a real thing and hasn't been for a while... though it looks like it might be by 10.1 so whatever).  I'm not sure why the Windows reactors are presented as "Win32 (....)" though.  How about "WFMO for Win32"?  Or even "WaitForMultipleObjects for Win32".  (Likewise for the headings for each of these reactors below)
    1. The "Mac OS X" reactor is really the cfreactor, which probably doesn't really work.  But we can at least mention its actual name here too.
    1. There's still a mention of GTK+ 1.2 in the "graphical toolkits" list
    1. The wxpython stuff is all in the "Non-Reactor GUI Integration" section, but `wxreactor` is a real reactor.  Some re-organization may make sense here.
    1. The `wxreactor` example is missing the `from twisted.internet import reactor` line which the other examples mostly have; it should be there.

Feel free to fix these here or file more tickets. :)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess commented|
|-|-|

(In [29017]) Don't consider poll a general-purpose reactor since it doesn't work on
OS X, and move the wxreactor description out of Non-Reactor GUI
Integration and into GUI Integration Reactors.

refs [#4384](https://github.com/twisted/twisted/issues/4384)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess removed owner|
|-|-|

Thanks for the review, exarkun. All of your review points should be addressed.

 * I moved poll to the platform-specific section. The supported Twisted platforms that support poll that aren't OS X are just Linux, right?
 * Unless Tkinter is special, it was also missing a reactor import.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_7"></a>|@therve set owner to @jesstess|
|-|-|

Looks good, please merge!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve commented|
|-|-|

Don't forget the NEWS fragement (but I don't think you can).

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_9"></a>|@jesstess set status to `closed`|
|-|-|

(In [29214]) Merge choosing-reactor-links-4384

Author: jesstess
Reviewers: exarkun, therve
Fixes: [#4384](https://github.com/twisted/twisted/issues/4384)

Update the choosing a reactor documentation to fix broken links and
reflect the current states of the various reactors, including
indicating that only select is a truly cross-platform reactor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
